src/views/Home/Home.tsx

```diff
@@ -1,88 +1,27 @@
-import { fetchTags } from "features/tag/tagSlice";
-
-import NavBar from "components/organisms/NavBar/NavBar";
 import SideBar from "components/organisms/SideBar/SideBar";
-import Modals from "components/organisms/Modals/Modals";
-import FilteredContainer from "components/organisms/Containers/FilteredContainer/FilteredContainer";
-import CalendarContainer from "components/organisms/Containers/CalendarContainer/CalendarContainer";
-import TagsContainer from "components/organisms/Containers/TagsContainer/TagsContainer";
-import TaskContainer from "components/organisms/Containers/TaskContainer/TaskContainer";
+import MainLayout from "layouts/MainLayout";
 
-import { getTodayString, getTomorrowString } from "utils/helpers";
-import * as S from "./styles";
+import { useUser } from "lib/auth/";
 
 const Home = () => {
-  const [isFetching, setIsFetching] = useState(true);
-  const [sideBarVisible, setSideBarVisible] = useState(false);
-  const dispatch = useAppDispatch();
   const navigate = useNavigate();
 
-  const handleToggleSideBar = () => {
-    setSideBarVisible(!sideBarVisible);
-  };
-
-  const tomorrowFilters = {
-    date: getTomorrowString(),
-  };
-
-  const todayFilters = {
-    date: getTodayString(),
-  };
+  const { user, isLoading } = useUser();
 
   useEffect(() => {
-    setIsFetching(true);
-    const token = localStorage.getItem("jwt");
-    if (token) {
-      const decoded: { userId: string; email: string; userName: string } =
-        jwt_decode(token);
-      dispatch(setUserName(decoded.userName));
-      dispatch(setEmail(decoded.email));
-      dispatch(fetchTags());
-
-      dispatch(fetchListArray()).then(() => {
-        setIsFetching(false);
-      });
-    } else {
+    if (!isLoading && !user) {
       navigate("/login");
     }
+
     //eslint-disable-next-line
   }, []);
 
-  return (
-    <div className="w-full ">
-      <NavBar handleToggleSideBar={handleToggleSideBar} />
-      <div className="w-full mt-14 ">
-        <SideBar isVisible={sideBarVisible} />
-        {!isFetching && (
-          <S.MainContainer $isVisible={sideBarVisible}>
-            <Routes>
-              <Route index element={<TaskContainer />} />
-              <Route
-                path="today"
-                element={<FilteredContainer filters={todayFilters} />}
-              />
-              <Route
-                path="tomorrow"
-                element={<FilteredContainer filters={tomorrowFilters} />}
-              />
-              <Route path="filtered" element={<FilteredContainer />} />
-              <Route path="calendar" element={<CalendarContainer />} />
-              <Route path="tags-and-filters" element={<TagsContainer />} />
-              <Route path=":listId" element={<TaskContainer />} />
-            </Routes>
-          </S.MainContainer>
-        )}
-      </div>
-      <Modals />
-    </div>
-  );
+  return !isLoading ? (
+    <MainLayout SideBar={SideBar}>{<Outlet />}</MainLayout>
+  ) : null;
```

Comment:
```suggestion
  return !isLoading && (
    <MainLayout SideBar={SideBar}>{<Outlet />}</MainLayout>
  );
```

Reply:
I wanted to do that but it returns an error "return type 'false | Element' is not a valid JSX element"